src/clj/fluree/db/api/transact.cljc

```diff
@@ -15,25 +15,32 @@
             [fluree.db.ledger :as ledger]))
 
 (defn parse-opts
-  [expanded-txn opts txn-context]
+  [expanded-txn override-opts txn-context]
   (let [txn-opts (some-> (util/get-first-value expanded-txn const/iri-opts)
                          util/keywordize-keys)
-        opts*    (merge txn-opts (util/keywordize-keys opts))]
-    (assoc opts* :context txn-context)))
+        opts*    (merge txn-opts (util/keywordize-keys override-opts))]
```

Comment:
I am not sure if it's worth bringing in the extra dependency, but it would be nice if we didn't have to do the ceremony of checking all different cases like we do in the `track-fuel?` function below, and [camel-snake-kebab](https://clj-commons.org/camel-snake-kebab/) would help with that.